src/Planners/include/BipedalLocomotion/Planners/UnicycleTrajectoryPlanner.h

```diff
@@ -0,0 +1,234 @@
+/**
+ * @file UnicycleTrajectoryPlanner.h
+ * @authors Lorenzo Moretti, Diego Ferigo, Giulio Romualdi, Stefano Dafarra
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+
+#include <BipedalLocomotion/Contacts/ContactPhaseList.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+#include <deque>
+#include <iDynTree/Model.h>
+#include <iDynTree/VectorDynSize.h>
+
+#include <memory>
+
+#include <UnicycleGenerator.h>
+#include <UnicyclePlanner.h>
+
+#include <CoMHeightTrajectoryGenerator.h>
+#include <vector>
+
+namespace BipedalLocomotion::Planners
+{
+class UnicycleTrajectoryPlanner;
+struct UnicycleTrajectoryPlannerInput;
+struct UnicycleTrajectoryPlannerOutput;
+struct UnicycleTrajectoryPlannerParameters;
+} // namespace BipedalLocomotion::Planners
+
+struct BipedalLocomotion::Planners::UnicycleTrajectoryPlannerInput
+{
+    /*
+    if UnicycleController::PERSON_FOLLOWING, the plannerInput is a vector of size 2 (i.e., [x, y])
+    if UnicycleController::DIRECT, the plannerInput is a vector of size 3 (i.e., [xdot, ydot, wz])
+    */
```

Comment:
I would add this comment in a more structured way. For instance 
```suggestion
/**
 * UnicycleTrajectoryPlannerInput implements the input to the planner. Depending on the type of unicycle controller used the plannerInput is a 2d-vector or a 3d-vector. For instance....
 */ 
struct BipedalLocomotion::Planners::UnicycleTrajectoryPlannerInput
{
   
```

Reply:
Done with [dfef4c1](https://github.com/ami-iit/bipedal-locomotion-framework/pull/844/commits/dfef4c1c3d56178c00c33f0b4069d362669ae101)